doc/modules/manifold.rst

```diff
@@ -513,12 +515,33 @@ divergence could increase during this phase. Usually it does not have to be
 tuned. A critical parameter is the learning rate. If it is too low gradient
 descent will get stuck in a bad local minimum. If it is too high the KL
 divergence will increase during optimization. More tips can be found in
-Laurens van der Maaten's FAQ (see references).
+Laurens van der Maaten's FAQ (see references). The last parameter, angle,
+is a tradeoff between performance and accuracy. Larger angles imply that we
+can approximate larger regions by a single point,leading to better speed
+but less accurate results. 
 
-Standard t-SNE that has been implemented here is usually much slower than
+The Barnes-Hut t-SNE that has been implemented here is usually much slower than
 other manifold learning algorithms. The optimization is quite difficult
-and the computation of the gradient is on :math:`O[d N^2]`, where :math:`d`
-is the number of output dimensions and :math:`N` is the number of samples.
+and the computation of the gradient is :math:`O[d N log(N)]`, where :math:`d`
+is the number of output dimensions and :math:`N` is the number of samples. The 
+Barnes-Hut method improves on the exact method where t-SNE complexity is 
+:math:`O[d N^2]`, but has several other notable differences:
+
+* The Barnes-Hut implementation only works when the target dimensionality is 3
+  or less. The 2D case is typical when building visualizations.
+* Barnes-Hut only works with dense input data. Sparse data matrices can only be
+  embedded with the exact method.
```

Comment:
"... or can be approximated by a dense low rank projection for instance using `sklearn.decomposition.TruncatedSVD`."

Please use the `:class:` sphinx reference  format to make have a link to the `TruncatedSVD` API.